templates/display.html

```diff
@@ -1,21 +1,39 @@
-{% extends 'base-deprecated.html' %}
+{% extends 'base.html' %}
 {% load static %}
 {% block content %}
 
 <div class="text-center">
-  {% if search %}
+  {% if saved %}
     {% if posts.count == 0 %}
-      <h4>There are no Syllabi matching your search at the {{school}} sorry! <br>Try to add them!</h4>
+      <h4>You have no saved syllabi sorry!<br>Try to save some!</h4>
     {% else %}
-      <h4>There {{posts.count|pluralize:"is,are"}} {{posts.count}} Syllab{{posts.count|pluralize:"us,i"}} matching your search at the {{school}}!</h4>
+      <h4>You have {{posts.count}} Syllab{{posts.count|pluralize:"us,i"}} saved!</h4>
+    {% endif %}
+  {% elif search %}
+    {% if posts.count == 0 %}
+      <h4>There are no Syllabi matching your search sorry! <br>Try to add some!</h4>
+    {% else %}
+      <h4>There {{posts.count|pluralize:"is,are"}} {{posts.count}} Syllab{{posts.count|pluralize:"us,i"}} matching your search!</h4>
     {% endif %}
   {% else %}
     <h4>There {{posts.count|pluralize:"is,are"}} {{posts.count}} Syllab{{posts.count|pluralize:"us,i"}} for {{dept}} courses! Add More!</h4>
   {% endif %}<br>
-
   {% for post in posts %}
-  <a href="https://{{AWS_S3_CUSTOM_DOMAIN}}/{{post.syllabus}}" class="btn btn-success" target="_blank">{{ post.dept }} {{ post.number }} taught by {{post.prof}}</a>
-  <br><br>
+    <form method="POST">
+      {% csrf_token %}
+      <input name="pk" value="{{post.pk}}" type="hidden"/>
+      <a href="https://{{AWS_S3_CUSTOM_DOMAIN}}/{{post.syllabus}}" class="btn btn-success" target="_blank">{{ post.dept }} {{ post.number }} taught by {{post.prof}}
+      {% if post in user.profile.saved.all %}
+      <button type="submit" name="unsave" value="Unsave" class="btn btn-sm">
+        <img src="{% static 'saved.png' %}"/>
+      </button>
+      {% else %}
+      <button type="submit" name="save" value="Save" class="btn btn-sm">
+        <img src="{% static 'unsaved.png' %}"/>
+      </button>
+      {% endif %}
```

Comment:
Move this out of the `<a>`, and make them `btn-link`s rather than `btn-sm`s.

Also give them `title` attributes like "Save for later", so that you can hover over the icon and figure out what it's for.

![image](https://user-images.githubusercontent.com/2766036/86052350-d8f8e280-ba24-11ea-8299-d317a5944350.png)

Reply:
Oh, it'll also need `alt`s on the imgs for accessibility, but you can probably just re-use the button's `title`.